### ISSUE TITLE
Pylint Commit Pre-hook

### ISSUE THREAD

**Issue by rbharath** [pandegroup/vs-utils#85]
I've been experimenting with using pylint as a git commit hook (see https://git-scm.com/book/es/v2/Customizing-Git-Git-Hooks). The pylint hook (see http://git-pylint-commit-hook.readthedocs.org/en/latest/) forces python code to pass pylint before accepting commits to git. Having to pass pylint made me clean large swathes of ugliness in my code that I'd been avoiding for weeks.

I propose that we make pylint commit hooks standard for all commits to vs-utils. I think the codebase will become significantly cleaner if we're forced to pass lint before commits and PRs.

Thoughts or comments?

**Comment by skearnes** [comment]
SGTM; I use something similar in the Travis builds for many of my other repos. I usually include a pyflakes check as well as pylint. Is the exception for Google-style spacing easy to add?

**Comment by rbharath** [comment]
Yep, the exception is easy to add as well. I'll make a PR that includes the config for the commit hook and update the readme with directions.

I like the idea of making a Travis build for this repo as well. If you're ok with it, I'll go ahead and set up a Travis build for vs-utils.

**Comment by skearnes** [comment]
Some of the tests won't work without some extensive setup via binstar or
others and may need to be excluded from the build (like AmberTools), but a
build that covers at least a subset of the tests is a good idea. You should
be able to use miniconda/binstar for the rdkit stuff I hope.
